isofit/utils/template_construction.py

```diff
@@ -1821,3 +1897,105 @@ def write_wavelength_file(filename, wl, fwhm):
         axis=1,
     )
     np.savetxt(filename, wl_data, delimiter=" ")
+
+
+def make_surface_config(
+    paths: Pathnames,
+    surface_category="multicomponent_surface",
+    pressure_elevation=None,
+    elevation_lut_grid=[],
+    surface_mapping: dict = None,
+    use_superpixels=False,
+):
+    """
+    Constructs the surface component of the config
+    Args:
+        paths: Pathnames object with all key values passed from apply_oe
+        surface_category: Base surface category
+    Returns:
+        surface_config_dict: Dictionary with all surface parameters and file
+                             locations.
+    """
+
+    # Initialize config dict
+    surface_config_dict = {
+        "multi_surface_flag": False,
+    }
+
+    # Check to see if a classification file is being propogated
+    # If so, use multisurface
+    if paths.surface_class_working_path:
+        surface_config_dict["Surfaces"] = {}
+
+        if use_superpixels:
+            surface_config_dict["surface_class_file"] = paths.surface_class_subs_path
+        else:
+            surface_config_dict["surface_class_file"] = paths.surface_class_working_path
+
+        surface_config_dict["base_surface_class_file"] = (
+            paths.surface_class_working_path
+        )
+
+        surface_config_dict["multi_surface_flag"] = True
+        surface_class_ds = envi.open(envi_header(paths.surface_class_working_path))
+
+        # Get the class mapping. Tried to build in some insensitivity here.
+        # Will use first option it hits
+        options = [
+            "mapping",
+            "Mapping",
+            "class names",
+            "Class names",
+            "class",
+            "Class",
+            "surfaces",
+            "Surfaces",
+            "surface names",
+            "Surface names",
+        ]
+        class_mapping = None
+        for option in options:
+            if class_mapping:
+                continue
```

Comment:
Per discussion, the suggestion is to enforce a standardized dictionary mapping between given index values and particular surface state types.